cumulusci/tasks/connectedapp.py

```diff
@@ -0,0 +1,174 @@
+import json
+import re
+import shutil
+import os
+import tempfile
+from cumulusci.core.config import ServiceConfig
+from cumulusci.core.exceptions import TaskOptionsError, ServiceNotConfigured
+from cumulusci.core.utils import process_bool_arg
+from cumulusci.tasks.sfdx import SFDXBaseTask, SFDX_CLI
+from cumulusci.utils import random_alphanumeric_underscore
+
+CONNECTED_APP = """<?xml version="1.0" encoding="UTF-8"?>
+<ConnectedApp xmlns="http://soap.sforce.com/2006/04/metadata">
+    <contactEmail>{email}</contactEmail>
+    <label>{label}</label>
+    <oauthConfig>
+        <callbackUrl>http://localhost:8080/callback</callbackUrl>
+        <consumerKey>{client_id}</consumerKey>
+        <consumerSecret>{client_secret}</consumerSecret>
+        <scopes>Web</scopes>
+        <scopes>Full</scopes>
+        <scopes>RefreshToken</scopes>
+    </oauthConfig>
+</ConnectedApp>"""
+
+PACKAGE_XML = """<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    <types>
+        <members>*</members>
+        <name>ConnectedApp</name>
+    </types>
+    <version>44.0</version>
+</Package>"""
+
+
+class CreateConnectedApp(SFDXBaseTask):
+    task_options = {
+        "label": {
+            "description": "The label for the connected app.  Must contain only alphanumeric and underscores",
+            "required": True,
+        },
+        "email": {
+            "description": "The email address to associate with the connected app.  Defaults to email address from the github service if configured.",
+            "required": True,
+        },
+        "username": {
+            "description": "Create the connected app in a different org.  Defaults to the defaultdevhubusername configured in sfdx.",
+            "required": False,
+        },
```

Comment:
This seems a bit odd since for all other CCI tasks we target orgs using the `--org` param. Use that if passed and otherwise use the dev hub?

Reply:
its a bootstrapping problem, right? you can't connect your devhub as a cci `--org` without a connected app.

Reply:
There's a chicken and egg type problem with this task in particular.  You would only want to create the connected app in a persistent org but you can't have a persistent org in the cci keychain until you have the connected app.  That's why this defaults to using an org from the sfdx keychain.

As a future enhancement, I might want to add another option, `prompt_password` which would prompt the user for a password and security token instead of using an org from the sfdx keychain.